src/Agent/Communication/AbstractRequest.php

```diff
@@ -323,7 +331,7 @@ public function addError($message, $code = 500)
             ),
             'expiration' => self::DEFAULT_FREQUENCY
             ]);
-        } else {
+        } else if ($code != 405) { // No need to add content on 405 error
```

Comment:
Maybe a `$add_error_to_response = true` parameter should be added in method signature. This would permit to use same behaviour for other result codes.

Reply:
With the latest commits, I set message to `null` in the case of 405 error and then won't insert a message checking if `$message` is empty or not.

Reply:
The c6ea1d2 commit remove this check and better handles the case in getResponse() by returning empty content if not content has been set.